src/parser/ast/ReturnStatementNode.h

```diff
@@ -65,10 +65,12 @@ class ReturnStatementNode : public StatementNode {
                 m_argument->generateExpressionByteCode(codeBlock, context, index);
 #endif
                 codeBlock->pushCode(ReturnFunctionSlowCase(ByteCodeLOC(m_loc.index), index), context, this->m_loc.index);
+                context->m_needsExtendedExecutionState = true;
             } else {
                 index = context->getRegister();
                 codeBlock->pushCode(LoadLiteral(ByteCodeLOC(m_loc.index), index, Value()), context, this->m_loc.index);
                 codeBlock->pushCode(ReturnFunctionSlowCase(ByteCodeLOC(m_loc.index), index), context, this->m_loc.index);
+                context->m_needsExtendedExecutionState = true;
```

Comment:
Is this marking in return statement also necessary?

Reply:
ReturnFunctionSlowCase opcode may needs ControlFlowRecord